src/coins.h

```diff
@@ -244,33 +250,24 @@ class CCoins
         }
         // coinbase height
         ::Unserialize(s, VARINT(nHeight), nType, nVersion);
-        Cleanup();
-    }
 
-    //! mark a vout spent
-    bool Spend(uint32_t nPos);
+        unsigned int bwtOutputCounter = 0;
+        if (this->IsFromCert()) {
+            ::Unserialize(s, originScId,       nType,nVersion);
+            ::Unserialize(s, bwtOutputCounter, nType,nVersion);
+        }
 
-    //! check whether a particular output is still available
-    bool IsAvailable(unsigned int nPos) const {
-        return (nPos < vout.size() && !vout[nPos].IsNull());
-    }
+        for(unsigned int idx = 0; idx < vout.size(); ++idx)
+            if ( idx < bwtOutputCounter)
```

Comment:
Pls check that the order of out type (from cert or not) in vout is the same as in CScCertificate class

Reply:
Done